source/custom/visa_service.custom.cpp

```diff
@@ -39,6 +43,275 @@ static ViSession GetResourceManagerSession(visa_grpc::VisaService::LibraryShared
   return s_defaultRM;
 }
 
+static AttributeValueData::DataCase GetAttributeType(ViAttr attributeID)
```

Comment:
I take it that we're not likely to add new attributes to visa? This would be hard to keep up-to-date in custom code. We could probably figure out how to generate it if necessary.

Reply:
Or perhaps a test that loops over all attributes in the proto

Reply:
Auto generation from `attributes.py` sounds more promising. If we have that, I don't think a test makes that much sense.

Reply:
Once you get the .mako hooked in, auto generation from our dictionary turns out to be quite easy. Thanks for the suggestion.